### PR TITLE
Improve ess_scalinggroup use common method

### PR DESCRIPTION
**Pull request by YY-tomorrow**

### CODE REVIEW

File: alicloud/data_source_alicloud_ess_scalinggroups.go [hunk 1]
```diff
@@ -129,34 +129,35 @@ func dataSourceAlicloudEssScalingGroups() *schema.Resource {
...
+		allScalingGroups = append(allScalingGroups, response.ScalingGroups.ScalingGroup...)
 
-		if len(resp.ScalingGroups.ScalingGroup) < PageSizeLarge {
+		if len(response.ScalingGroups.ScalingGroup) < PageSizeLarge {
 			break
 		}
 
-		if page, err := getNextpageNumber(args.PageNumber); err != nil {
+		if page, err := getNextpageNumber(request.PageNumber); err != nil {
 			return err
```
Comment (ddaddaddda) [inline alicloud/data_source_alicloud_ess_scalinggroups.go]: 158  WrapError(err)

File: alicloud/resource_alicloud_ess_scalingconfiguration.go [hunk 2]
```diff
@@ -443,7 +443,7 @@ func enableEssScalingConfiguration(d *schema.ResourceData, meta interface{}) err
 				if err != nil {
 					return fmt.Errorf("EnableScalingGroup %s got an error: %#v", sgId, err)
 				}
-				if err := essService.WaitForScalingGroup(sgId, Active, DefaultTimeout); err != nil {
+				if err := essService.WaitForEssScalingGroup(sgId, Active, DefaultTimeout); err != nil {
 					return fmt.Errorf("WaitForScalingGroup is %#v got an error: %#v.", Active, err)
```
Comment (ddaddaddda) [inline alicloud/resource_alicloud_ess_scalingconfiguration.go]: return WrapError(err)

File: alicloud/resource_alicloud_ess_scalingconfiguration.go [hunk 3]
```diff
@@ -459,7 +459,7 @@ func enableEssScalingConfiguration(d *schema.ResourceData, meta interface{}) err
 				if err != nil {
 					return fmt.Errorf("DisableScalingGroup %s got an error: %#v", sgId, err)
 				}
-				if err := essService.WaitForScalingGroup(sgId, Inactive, DefaultTimeout); err != nil {
+				if err := essService.WaitForEssScalingGroup(sgId, Inactive, DefaultTimeout); err != nil {
 					return fmt.Errorf("WaitForScalingGroup is %#v got an error: %#v.", Inactive, err)
```
Comment (ddaddaddda) [inline alicloud/resource_alicloud_ess_scalingconfiguration.go]: 同上

File: alicloud/resource_alicloud_ess_scalinggroup.go [hunk 4]
```diff
@@ -178,92 +179,100 @@ func resourceAliyunEssScalingGroupRead(d *schema.ResourceData, meta interface{})
 func resourceAliyunEssScalingGroupUpdate(d *schema.ResourceData, meta interface{}) error {
 
 	client := meta.(*connectivity.AliyunClient)
-	args := ess.CreateModifyScalingGroupRequest()
-	args.ScalingGroupId = d.Id()
-
-	d.Partial(true)
```
Comment (ddaddaddda) [inline alicloud/resource_alicloud_ess_scalinggroup.go]: 184  我觉得 d.Partial(true) 不能去掉，因为create后还在调用update

File: alicloud/service_alicloud_ess.go [hunk 5]
```diff
@@ -319,7 +320,7 @@ func (srv *EssService) EssRemoveInstances(groupId string, instanceIds []string)
 	if group.LifecycleState == string(Inactive) {
 		return fmt.Errorf("Scaling group current status is %s, please active it before attaching or removing ECS instances.", group.LifecycleState)
```
Comment (ddaddaddda) [inline alicloud/service_alicloud_ess.go]: 321  WrapError

File: alicloud/resource_alicloud_ess_attachment_test.go [hunk 6]
```diff
@@ -81,7 +81,7 @@ func testAccCheckEssAttachmentDestroy(s *terraform.State) error {
 			continue
 		}
 
-		_, err := essService.DescribeScalingGroup(rs.Primary.ID)
+		_, err := essService.DescribeEssScalingGroup(rs.Primary.ID)
 		if err != nil {
 			if NotFoundError(err) || IsExceptedError(err, InvalidScalingGroupIdNotFound) {
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_ess_attachment_test.go]: I think `IsExceptedError(err, InvalidScalingGroupIdNotFound)` is redundant.

File: alicloud/resource_alicloud_ess_scalinggroup.go [hunk 7]
```diff
@@ -178,92 +179,106 @@ func resourceAliyunEssScalingGroupRead(d *schema.ResourceData, meta interface{})
...
-	_, err := client.WithEssClient(func(essClient *ess.Client) (interface{}, error) {
-		return essClient.ModifyScalingGroup(args)
+	raw, err := client.WithEssClient(func(essClient *ess.Client) (interface{}, error) {
+		return essClient.ModifyScalingGroup(request)
 	})
 	if err != nil {
-		return err
+		return WrapErrorf(err, DefaultErrorMsg, d.Id(), request.GetActionName(), AlibabaCloudSdkGoERROR)
 	}
-
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_ess_scalinggroup.go]: I think there is no need to `SetPartial `

**Pull request merged**